python/tank/platform/engine.py

```diff
@@ -970,6 +970,8 @@ def register_command(self, name, callback, properties=None):
                 # we know the app for the existing item.
                 # so prefix with app name
                 prefix = existing_item["properties"].get("app").instance_name
+                if existing_item["properties"].get("group"):
+                    prefix = "%s:%s" % (prefix, existing_item["properties"]["group"])
```

Comment:
General comments:

- It seems the code is doing some sort of two pass thing using a member variable `self.__commands_that_need_prefixing` - can this be cleaned up and simplified somehow? It looks complicated.

- We need more comments in the code to both explain the the case when two app instances register the same command and the command key therefore gets prefixed with the app instance, and the case that you just added with grouping.

Can we also add unit tests to this? I suspect there may not be any unit tests for register_command yet - now is a  good time to add them. With any change to core, we should make sure it's covered by tests unless we have good reasons not to.